drivers/sensor/ak8975/ak8975.c

```diff
@@ -119,28 +118,25 @@ static int ak8975_read_adjustment_data(struct ak8975_data *drv_data)
 
 int ak8975_init(const struct device *dev)
 {
-	struct ak8975_data *drv_data = dev->data;
+	const struct ak8975_config *drv_config = dev->config;
 	uint8_t id;
 
-	drv_data->i2c =
-		device_get_binding(DT_INST_BUS_LABEL(0));
-	if (drv_data->i2c == NULL) {
-		LOG_ERR("Failed to get pointer to %s device!",
-			    DT_INST_BUS_LABEL(0));
-		return -EINVAL;
+	if (!device_is_ready(drv_config->i2c.bus)) {
+		LOG_ERR("I2C bus device not ready");
+		return -ENODEV;
 	}
 
 #if DT_NODE_HAS_STATUS(DT_INST(0, invensense_mpu9150), okay)
 	/* wake up MPU9150 chip */
-	if (i2c_reg_update_byte(drv_data->i2c, MPU9150_I2C_ADDR,
+	if (i2c_reg_update_byte(drv_config->i2c.bus, MPU9150_I2C_ADDR,
```

Comment:
Why isn't this changed to `i2c_reg_update_byte_dt`?

Reply:
To my understandig it's another device on the same bus but with a different address. Added a TODO in the commit message about moving mpu9150 to its own driver. Seems to me as if a very specific configuration of two sensors has made it into the driver.

Reply:
I think that we should update code so that the MPU9150 address is picked from DT. This IC is a bit weird, it embeds 2 sensors inside (they probably took 2 sensors and placed them on the same die). The _main_ one (acc/gyro) is being used here. the magnetometer, AK8975, is the other one attached to the same bus. For the magnetometer, address could be hardcoded (I think it's fixed?) or taken from a new property like `reg-mag`.

Reply:
Should it be addressed in this pull request?